var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -103,4 +103,31 @@ def cmake_args(self):
                 '-DCMAKE_CXX_FLAGS=-DGLX_GLXEXT_LEGACY'
             ])
 
+            # VTK 6.1.0 (and possibly earlier) does not use
+            # NETCDF_CXX_ROOT to detect NetCDF C++ bindings, so
+            # NETCDF_CXX_INCLUDE_DIR and NETCDF_CXX_LIBRARY must be
+            # used instead to detect these bindings
+
+            netcdf_cxx_lib = find_libraries('libnetcdf_c++',
```

Comment:
please move it to the `netcdf` package and use `libs()` interface, see [hdf5](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/hdf5/package.py#L103-L164) on how to use this.

Reply:
I agree with what you're saying, and I think the modifications you're asking for require a separate pull request.

Here's what would be required:

- deleting the `netcdf-cxx`, `netcdf-cxx4`, `netcdf-fortran`, and possibly the `py-netcdf` packages
- vendoring the deleted packages in as resources
- rewriting the build instructions entirely for the deleted packages, because the `AutotoolsPackage` infrastructure can no longer be leveraged; each resource will have to be staged, configured, built, and installed

Depending on the bugs encountered, that process could take a couple days (because I'll need to incrementally rebuild NetCDF several times with these interfaces), and I may not be able to spare that time for a few weeks, because I'm on deadline and actually need to get a VisIt build working now -- I have another PR just for that, and it depends on this PR and PR#6943. The combination of this PR, PR#6943, and the upcoming VisIt PR will resolve VisIt issues that have been open for months. Do you really want to hold up this PR until I do that reorganization? If so, do you want me to include the Fortran and Python interfaces in that reorganization as well?

Reply:
> deleting the netcdf-cxx, netcdf-cxx4, netcdf-fortran, and possibly the py-netcdf packages

i think you over-complicate things. It was probably mis-leading on my side to say

> move it to the netcdf package and use libs() interface

what I meant is to move it to `netcdf-cxx`.

 You don't need to combine all the `netcdf` packages. That's too much work and I think at some point we decided to keep them separate as they have separate versions.

Reply:
p.s. essentially you just need a custom `libs()` function for `netcdf-cxx` as the default one does not work here since libs are called `libnetcdf_c++`.

Reply:
That `libs` method already exists for `netcdf-cxx`, so now I build up the path using `spec['netcdf-cxx'].libs().joined()`.

Reply:
ah, good, i did not notice.

Reply:
I still need to test the new changes; working on finalizing the VisIt PR.